client-common/src/storage.rs

```diff
@@ -128,17 +134,21 @@ where
         passphrase: &SecUtf8,
     ) -> Result<Option<Vec<u8>>> {
         let old_value = self.get_secure(&keyspace, &key, passphrase)?;
-
-        let mut algo = get_algo(passphrase);
-
         let mut nonce = [0u8; NONCE_SIZE];
+        let mut salt = [0u8; SALT_SIZE];
         OsRng.fill(&mut nonce);
-
-        let mut cipher = algo.seal(&nonce, key.as_ref(), &value);
+        OsRng.fill(&mut salt);
+        let algo = get_algo(passphrase, &salt)?;
+        let mut cipher: Vec<u8> = algo
+            .encrypt(
+                &GenericArray::clone_from_slice(&nonce),
```

Comment:
to get a bit better static checking, I think one could use `.into()`: https://docs.rs/generic-array/0.12.3/generic_array/struct.GenericArray.html
There are all these `Into` implementations for different fixed array sized (and here one uses the fixed size `[0u8; NONCE_SIZE]`)